core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/DetailedSampleBuilder.java

```diff
@@ -996,4 +981,19 @@ public String getBarcodeSizeInfo() {
     return "HOW DID YOU EVER PRINT THIS?!?";
   }
 
+  @Override
+  public String getDeleteType() {
+    throw new UnsupportedOperationException("Cannot delete non-persisted builder object");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Cannot delete non-persisted builder object" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)